pertpy/tools/_perturbation_space/_simple.py

```diff
@@ -59,9 +62,106 @@ def __call__(self, adata: AnnData, embedding_key: str = "X_umap", *args, **kwarg
         return ps_adata
 
 
-class PseudobulkSpace:
+class PseudobulkSpace(PerturbationSpace):
     """Determines pseudobulks of an AnnData object."""
 
-    def __call__(self, *args, **kwargs):
-        # TODO implement
-        pass
+    def __call__(self, adata: AnnData, target_col: str = "perturbations", layer_key: str = None, **kwargs):  # type: ignore
+        # Create a DataFrame from the .obs attribute of the AnnData object
+        obs_df = adata.obs
+        # Group the observations based on the 'condition' column
+        grouped = obs_df.groupby(target_col)
+
+        if layer_key:
+            X = np.empty((len(adata.obs[target_col].unique()), adata.layers[layer_key].shape[1]))
+            index = []
+            i = 0
+            for group_name, group_data in grouped:
+                indices = group_data.index
+                index.append(group_name)
+                X[i, :] = np.mean(adata[indices].layers[layer_key], axis=0)
+                i += 1
+
+        else:
+            X = np.empty((len(adata.obs[target_col].unique()), adata.X.shape[1]))
+            index = []
+            i = 0
```

Comment:
I hate `i` as variable name. Can you make it more explicit?

Reply:
done